tf2_ros/include/tf2_ros/buffer.h

```diff
@@ -261,7 +261,11 @@ namespace tf2_ros
     CreateTimerInterface::SharedPtr timer_interface_;
 
     /// \brief A map from active timers to BufferCore request handles
-    std::unordered_map<TimerHandle, tf2::TransformableRequestHandle> timer_to_request_map_;
+    typedef struct {
+      tf2::TransformableRequestHandle request_handle;
+      tf2::TransformableCallbackHandle callback_handle;
+    } TransformableHandles;
+    std::unordered_map<TimerHandle, TransformableHandles> timer_to_request_map_;
```

Comment:
So this almost works, but it potentially changes the ABI by changing this private member variable.

To get around this, we need to figure out a way to keep this member variable the same but add in the functionality we want.  One way to do this is to leave the map as it is, but then add a second map (outside of the class) that maps between a `TransformableRequestHandle` and a `TransformableCallbackHandle`.  Then we could do a lookup in both maps to find what we need.  Note that if we go this way, we also need to add in locking, as it could be accessed from multiple threads simultaneously.

That's just one idea.  There may be other ways to fix this in an ABI-compatible way.

Reply:
Ah, understood. I created a seperate map to do just that. The locking is already done by the `timer_to_request_map_mutex_` so we don't have to worry about that.

But why is it important that this variable is not changed? It is a private variable so no other class will be able to access it.

Reply:
> It is a private variable so no other class will be able to access it.

Yep, that's right.  Because of this, changing it would still be API compliant.

> But why is it important that this variable is not changed? 

But the problem is that changing the type could change the *ABI*, and we guarantee that in stable releases as well.  That's why we can't change it here.